### DISCUSSION TITLE
Can modifiers automatically remove print statements before submit it

### DISCUSSION THREAD

**Discussion by rgbygv** [j178/leetgo#316] [Q&A] [answered]
If we neet a complex problem, we sometimes need to print information to debug it. But print statements in final submit code only slow down the code and we always fortget to remove these print statements, so I think it might be helpful to have a modifier that automatically removes the print statement but doesn't affect local debugging.

**Comment by j178** [comment, marked as answer]
This is a good idea; however, currently, modifiers only run during the code generation stage, not the submission stage. Implementing this might require some big refactoring.

**Reply by rgbygv** [reply]
Yes, and it needs to be handled in different languages.
In my first assumption, we just read the code line by line and if we find a print statement, we delete it,
but in reality, it is more complicated. And the code style may also affect the modifiers.
For example, if there are some codes like
```python
if some_condition:
    print(some_value)
else:
   print(other_value)
```
It is difficult to delete them.
Maybe it is not worth spending too much time on this feature.
